src/Common/Logging/EventAuditLogger.php

```diff
@@ -632,6 +640,27 @@ public function auditSQLEvent($statement, $outcome, $binds = null)
             }
         }
 
+        /**
+         * Avoid logging sensitive information in logs.
+         * For any insert operation, if the current table contains any field containing sensitive information,
+         * the value for that field is masked to match its length.
+         */
+        foreach (self::SENSITIVE_RECORDS as $table => $fields) {
+            if ($querytype == "insert") {
+                if (strpos($comments, $table) !== false) {
+                    foreach ($fields as $field) {
+                        $pattern = "/($field)\s*=\s*'([^']+)'/";
+                        $comments = preg_replace_callback($pattern, function ($matches) {
+                            $field_name = $matches[1];
+                            $field_value = $matches[2];
+                            $masked_value = str_repeat('X', strlen($field_value));
+                            return "$field_name = '$masked_value'";
+                        }, $comments);
```

Comment:
![bird_jive](https://github.com/openemr/openemr/assets/5618372/0f2010ca-f4e6-4fe9-a587-0ba934bfe772)

Reply:
@stephenwaite is there something else that needs to be updated to the PR?